src/MagnumPlugins/BasisImageConverter/Test/BasisImageConverterTest.cpp

```diff
@@ -0,0 +1,276 @@
+
+BasisImageConverterTest::BasisImageConverterTest() {
+    addTests({&BasisImageConverterTest::wrongFormat,
+              &BasisImageConverterTest::zeroSize,
+
+              &BasisImageConverterTest::r,
+              &BasisImageConverterTest::rg,
+              &BasisImageConverterTest::rgb,
+              &BasisImageConverterTest::rgba});
+
+    /* Pull in the AnyImageImporter dependency for image comparison, load
+       StbImageImporter from the build tree, if defined. Otherwise it's static
+       and already loaded. */
+    _manager.load("AnyImageImporter");
+    #ifdef PNGIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_importerManager.load(PNGIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+    /* Load the plugin directly from the build tree. Otherwise it's static and
+       already loaded. */
+    #ifdef BASISIMAGECONVERTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_converterManager.load(BASISIMAGECONVERTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+    #ifdef BASISIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_importerManager.load(BASISIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+}
+
+void BasisImageConverterTest::wrongFormat() {
+    Containers::Pointer<AbstractImageConverter> converter =
+        _converterManager.instantiate("BasisImageConverter");
+    ImageView2D image{PixelFormat::RG32F, {}, nullptr};
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!converter->exportToData(image));
+    CORRADE_COMPARE(out.str(), "Trade::BasisImageConverter::exportToData(): unsupported format PixelFormat::RG32F\n");
+}
+
+void BasisImageConverterTest::zeroSize() {
+    Containers::Pointer<AbstractImageConverter> converter =
+        _converterManager.instantiate("BasisImageConverter");
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!converter->exportToData(ImageView2D{PixelFormat::RGB8Unorm, {}, nullptr}));
+    CORRADE_COMPARE(out.str(),
+        "Trade::BasisImageConverter::exportToData(): source image is empty\n");
+}
+
+void BasisImageConverterTest::r() {
+    if(_importerManager.loadState("PngImporter") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("PngImporter plugin not found, cannot test contents");
+
+    Containers::Pointer<AbstractImporter> pngImporter =
+        _importerManager.instantiate("PngImporter");
+    pngImporter->openFile(Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb-27x63.png"));
+    const auto originalImage = pngImporter->image2D(0);
+
+    /* Create image with skip and padding for aligning rows to 4 bytes */
+    const Containers::Array<UnsignedByte> imageDataWithSkip{Containers::ValueInit,
+        ((27+8) + 1)*(63+7)};
+    const ImageView2D imageWithSkip{PixelStorage{}.setSkip({0, 1, 0}),
+        PixelFormat::R8Unorm, originalImage->size(), imageDataWithSkip};
+
+    auto sourcePixels = originalImage->pixels<Color4ub>();
```

Comment:
```suggestion
    auto sourcePixels = Containers::arrayCast<UnsignedByte>(originalImage->pixels<Color4ub>());
```

and then you can just assign without doing `.r()`. That's the "canonical" workflow -- have the loops as dumb as possible :)

Similar in the two cases below -- there it seems to still be the original range-loop.

Reply:
This leads to

~~~
D:\GitHub\vcpkg\installed\x64-windows\include\Corrade/Containers/StridedArrayView.h(823): error C2440: 'initializing': cannot convert from 'init
ializer list' to 'Corrade::Containers::StridedArrayView<2,Magnum::UnsignedByte>'
D:\GitHub\vcpkg\installed\x64-windows\include\Corrade/Containers/StridedArrayView.h(823): note: No constructor could take the source type, or co
nstructor overload resolution was ambiguous
..\src\MagnumPlugins\BasisImageConverter\Test\BasisImageConverterTest.cpp(136): note: see reference to function template instantiation 'Corrade:
:Containers::StridedArrayView<2,Magnum::UnsignedByte> Corrade::Containers::arrayCast<Magnum::UnsignedByte,2,const Magnum::Color3ub>(const Corrad
e::Containers::StridedArrayView<2,const Magnum::Color3ub> &)' being compiled
ninja: build stopped: subcommand failed.
~~~

Reply:
wtf

Reply:
`Containers::arrayCast<const UnsignedByte>(...` was the solution, just a missing `const`.